### PR TITLE
vagrant env updates to enable ceph config overrides and a good default for OSD database size

### PR DESCRIPTION
**Pull request by jbw976**

### CODE REVIEW

File: demo/vagrant/Vagrantfile [hunk 1]
```diff
@@ -58,6 +58,9 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
         node.vm.synced_folder "../../release", "/release", id: "release", :nfs => true, :mount_options => ['nolock,vers=3,tcp']
       end
 
+      # copy ceph config override file
+      node.vm.provision "file", source: "ceph-override.conf", destination: "/tmp/rook-conf/ceph-override.conf"
```
Comment (bassam) [inline demo/vagrant/Vagrantfile]: I think `/var/lib/rook` is a better location than /tmp

Comment (jbw976) [inline demo/vagrant/Vagrantfile]: i wanted vagrant to not get in the way of `rookd` making/owning that directory, but i'll give it a shot.  it would be cleaner to just use `/var/lib/rook` as that would remove the extra bind mount too.

Comment (travisn) [inline demo/vagrant/Vagrantfile]: what if you put this in the user folder instead of /tmp?

File: demo/vagrant/cloud-config.yml.in [hunk 2]
```diff
@@ -7,6 +7,11 @@ coreos:
   - name: rookd.service
     enable: true
     command: start
+    drop-ins:
+      - name: 10-db-size.conf
+        content: |
+          [Service]
+          Environment=ROOKD_OSD_DATABASE_SIZE=1024
```
Comment (travisn) [inline demo/vagrant/cloud-config.yml.in]: should this be a separate PR or commit?

Comment (jbw976) [inline demo/vagrant/cloud-config.yml.in]: moved into its own commit and updated the title of the PR

File: demo/vagrant/Vagrantfile (hunk 1 above)
Comment (jbw976) [inline demo/vagrant/Vagrantfile]: actually changed the whole approach to use `write_files` directive in the cloud config.  i think it's pretty clean now.
